File: Typescript/Express/src/models/Product.ts
import {  ModelCtor,Model } from "sequelize/types";
import {DataTypes} from "sequelize";
import db from "../helpers/database/connectDatabase";
import IModel from "./IModel";

export interface IProduct extends IModel {
    categoryId:number;
    supplierId:number;
    productName:string;
    productDescription:string;
    unitPrice:number;
    unitsInStock:number;
}

const Product:ModelCtor<Model<IProduct | IModel,any>>= db.define('product',{
    categoryId:{
        type:DataTypes.INTEGER,
        field:"category_id"
    },
    supplierId:{
        type:DataTypes.INTEGER,
        field:"supplier_id"
    },
    productName: {
        type:DataTypes.STRING,
        field:"product_name"
    },
    productDescription: {
        type:DataTypes.STRING,
        field:"product_desc"
    },
    unitPrice: {
        type:DataTypes.DOUBLE,
        field:"unit_price"
    },
    unitsInStock: {
        type: DataTypes.DOUBLE,
        field:"units_in_stock"
    }
},{tableName:"products",createdAt:false,updatedAt:false,
    
})

export default Product;

File: CSharp/DotNetWebAPI/WebAPI/Entities/Product.cs
namespace WebAPI.Entities
{
    public class Product:IEntity
    {
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public int SupplierId { get; set; }
        public string ProductName { get; set; }
        public string ProductDescription { get; set; }
        public double UnitPrice { get; set; }
        public double UnitsInStock { get; set; }
    }
}